### PR TITLE
[13562] Wallet user avatar component

### LINKED ISSUE #13562: UI Component - Wallet User Avatar
This issue is to track the creation of the component: **Wallet User Avatar**

You can access the link at https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Mobile-Components?node-id=114%3A7646.

Please have a chat with the design team before starting with the component, and don't forget to add sizing to the issue

### PR DESCRIPTION
**Pull request by ibrkhalil**
fixes #13562 

status: ready

image

### CODE REVIEW

**Comment by Parveshdhull** [comment]
HI @ibrkhalil,  linting failed for the PR. You can run `make lint` to see the errors, and then you can fix them manually or run `make lint-fix`

**Comment by ibrkhalil** [comment]
> HI @ibrkhalil, linting failed for the PR. You can run `make lint` to see the errors, and then you can fix them manually or run `make lint-fix`

Can we use something like this ?
https://github.com/marketplace/actions/pre-commit
To run clj-kondo pre-commits, or is it overkill ?

File: src/quo2/components/wallet_user_avatar.cljs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                                                size :x-large}}]
+  (let [circle-size (size circle-sizes)
+        small? (= size :small)
+        f-name-initial (-> f-name
+                           clojure-string/upper-case
+                           (subs 0 1))
+        l-name-initial (-> l-name
+                           clojure-string/upper-case
+                           (subs 0 1))
+        is-theme-dark? (if dark? :dark :light)
```
Comment (flexsurfer) [inline src/quo2/components/wallet_user_avatar.cljs]: `is-theme-dark?` more looks like `theme`

**Review by flexsurfer — APPROVED** [review]

File: src/quo2/components/wallet_user_avatar.cljs (hunk 1 above)
Comment (ibrkhalil) [reply src/quo2/components/wallet_user_avatar.cljs]: Renamed

File: src/quo2/components/wallet_user_avatar.cljs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                 :small 11})
+
+(defn wallet-user-avatar
+  "params, first name, last name, color, size
+   and if it's dark or not!"
+  [{:keys [f-name l-name color dark? size] :or {f-name "John"
+                                                l-name "Doe"
+                                                color :red
+                                                dark? false
+                                                size :x-large}}]
```
Comment (Parveshdhull) [inline src/quo2/components/wallet_user_avatar.cljs]: we should only mock data in preview, component should be in its final usable form

File: src/quo2/components/wallet_user_avatar.cljs [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+(def circle-sizes {:small 20
+                   :medium 32
+                   :large 48
+                   :x-large 80})
+
+(def font-sizes {:x-large 27
+                 :large 15
+                 :medium 13
+                 :small 11})
```
Comment (Parveshdhull) [inline src/quo2/components/wallet_user_avatar.cljs]: :small -> :x-large
:x-large -> :small

(no strong opinion, but same order looks better, maybe from small to x-large)

File: src/quo2/foundations/colors.cljs [hunk 4]
```diff
@@ -183,6 +183,87 @@
...
+
+
+(def switcher-background "#040B14")
+
+;;switcher-screen with transparency
+(def switcher-background-opa-60 (alpha switcher-background 0.6))
+(def switcher-background-opa-70 (alpha switcher-background 0.7))
+(def switcher-background-opa-80 (alpha switcher-background 0.8))
+(def switcher-background-opa-90 (alpha switcher-background 0.9))
+(def switcher-background-opa-95 (alpha switcher-background 0.95))
```
Comment (Parveshdhull) [inline src/quo2/foundations/colors.cljs]: irrelevant change?

File: src/quo2/components/wallet_user_avatar.cljs [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+(defn wallet-user-avatar
+  "params, first name, last name, color, size
+   and if it's dark or not!"
+  [{:keys [f-name l-name color dark? size] :or {f-name "John"
```
Comment (Parveshdhull) [inline src/quo2/components/wallet_user_avatar.cljs]: `dark?` (please use app theme instead of parameter)

**Comment by siddarthkay** [comment]
@ibrkhalil : Design issue 1 

The background color in dark mode does not have correct opacities for color type primary.

Screenshot 2022-08-22 at 4 51 00 PM

File: src/quo2/components/wallet_user_avatar.cljs (hunk 3 above)
Comment (ibrkhalil) [reply src/quo2/components/wallet_user_avatar.cljs]: Done

File: src/quo2/components/wallet_user_avatar.cljs (hunk 2 above)
Comment (ibrkhalil) [reply src/quo2/components/wallet_user_avatar.cljs]: Done

File: src/quo2/components/wallet_user_avatar.cljs (hunk 5 above)
Comment (ibrkhalil) [reply src/quo2/components/wallet_user_avatar.cljs]: Done

File: src/quo2/foundations/colors.cljs (hunk 4 above)
Comment (ibrkhalil) [reply src/quo2/foundations/colors.cljs]: Done

**Pull request merged**